src/QsCompiler/CompilationManager/DataStructures.cs

```diff
@@ -176,49 +171,48 @@ public bool Equals(CodeFragment other)
                 return false;
             }
             return
-                this.GetRange().Equals(other.GetRange()) &&
+                this.Range == other.Range &&
                 this.Indentation == other.Indentation &&
                 this.Text == other.Text &&
                 this.FollowedBy == other.FollowedBy &&
                 this.IncludeInCompilation == other.IncludeInCompilation &&
                 (this.Kind == null ? other.Kind == null : this.Kind.Equals(other.Kind));
         }
 
-        internal CodeFragment WithUpdatedLineNumber(int lineNrChange) =>
-            this?.SetRange(this.GetRange().WithUpdatedLineNumber(lineNrChange));
+        internal CodeFragment TranslateLines(int offset) => this.SetRange(this.Range.TranslateLines(offset));
```

Comment:
TranslateLines to me is less descriptive than WithUpdatedLineNumber. What is translated? The former lets you know the only thing that changes is the line number.

Reply:
I had the opposite reaction with `WithUpdatedLineNumber`: how is it updated? I assumed that "updated" meant "replaced," like an F# `with` expression, but it actually meant "added to." I chose `TranslateLines` to try to capture the fact that it updates the line numbers (lines) by adding (translating) them by the given amount.

Reply:
For `Diagnostic.TranslateLines`, from @bettinaheim:
> Same here as above; TranslateLines is really cryptic to me. It does not modify any lines, or even have any lines; it has line numbers, and those get changed/updated.

For `Range.TranslateLines`, from @bettinaheim:
> Even though the naming here is clearer, I would still go with something like WithLineNrOffset.

Reply:
It seems like part of the problem is that I'm using lines and line numbers interchangeably. Would `TranslateLineNums` (I prefer `Num` over `Nr`) be better? Or is the issue also with "translating"? I am using translation in the geometric sense, so the lines (or line numbers) are moving by some amount.

Reply:
The translate to me is confusing. How about something like WithAddedLineNr or WithLineNrOffset? In any case, anything with line number in there is more understandable, and I'd not use translate.

Reply:
I have renamed `TranslateLines` to `WithLineNumOffset` in #579.

Reply:
Thank you!